src/dotnet/Chat.UI.Blazor/Services/ChatListUI.cs

```diff
@@ -82,32 +78,38 @@ public int GetCountWhenLoading(ChatListKind listKind)
         };
 
     [ComputeMethod]
-    public virtual Task<int> GetCount(ChatListKind listKind)
+    public virtual async Task<int> GetCount(PlaceId placeId, CancellationToken cancellationToken)
     {
-        var items = GetItems(listKind);
-        lock (items)
-            return Task.FromResult(items.Count);
+        var listView = await ActiveChatListView.Use(cancellationToken).ConfigureAwait(false);
+        if (listView == null)
```

Comment:
listView has ref to PlaceId. We can use it to quickly determine if count = 0 when given placeId does not correspone active list view placeId.